src/software/backend/input/network/network_filter.h

```diff
@@ -0,0 +1,127 @@
+#include "software/proto/messages_robocup_ssl_wrapper.pb.h"
+#include "software/proto/ssl_referee.pb.h"
+#include "software/sensor_fusion/refbox_data.h"
+#include "software/time/timestamp.h"
+#include "software/world/ball.h"
+#include "software/world/ball_state.h"
+#include "software/world/field.h"
+#include "software/world/team.h"
+
+class NetworkFilter
+{
+   public:
+    /**
+     * Creates a new NetworkFilter for data input and filtering
+     */
+    explicit NetworkFilter();
+
+    /**
+     * Returns a new Field object containing the most up to date state of the field given
+     * the new GeometryData information
+     *
+     * @param geometry_packet The SSL_GeometryData packet containing new field data
+     *
+     * @return a Field object containing the most up to date state of the field given the
+     * new GeometryData information
+     */
+    Field getFieldData(const SSL_GeometryData &geometry_packet);
+
+    /**
+     * Filters the ball data contained in the list of SSL detection frames and returns the
+     * ball detections
+     *
+     * @param detections A list of new DetectionFrames containing ball data
+     *
+     * @return all the ball detections contained in the ssl detection frames
+     */
+    std::vector<BallDetection> getBallDetections(
+        const std::vector<SSL_DetectionFrame> &detections);
+
+    /**
+     * Filters the robot data for the given team contained in the list of DetectionFrames
+     * and returns the most up to date detections of the given team
+     *
+     * @param detections A list of new DetectionFrames containing given team robot data
+     * @param team_type the team to get detections for
+     *
+     * @return The most up to date detections of the given team given the new
+     * DetectionFrame information
+     */
+    std::vector<RobotDetection> getTeamDetections(
+        const std::vector<SSL_DetectionFrame> &detections, TeamType team_type);
+
+    /**
+     * Parses RefboxData from a Referee proto message
+     *
+     * @param packet Referee proto to parse
+     *
+     * @return RefboxData parsed from the packet
+     */
+    RefboxData getRefboxData(const Referee &packet);
+
+    virtual ~NetworkFilter() = default;
+
+   private:
+    /**
+     * Creates a Field object given geometry data from a protobuf packet
+     *
+     * @param packet_geometry The SSL_GeometryFieldSize data from a protobuf packet
+     * containing field geometry
+     * @return A Field object representing the field specified with the provided geometry
+     */
+    Field createFieldFromPacketGeometry(
+        const SSL_GeometryFieldSize &packet_geometry) const;
+
+    // Objects used to aggregate and store state. We use these to aggregate the state
+    // so that we always publish "complete" data, not just data from a single frame/
+    // part of the field
+    Field field_state;
+    BallState ball_state;
+    Team friendly_team_state;
+    Team enemy_team_state;
+
+    BallFilter ball_filter;
```

Comment:
We shouldn't have filters in this class anymore

Reply:
bumping this comment

Reply:
done